src/Internal.AspNetCore.BuildTools.Tasks/ZipArchive.cs

```diff
@@ -86,8 +87,23 @@ public override bool Execute()
                         return false;
                     }
 
-                    zip.CreateEntryFromFile(file.ItemSpec, entryName);
-                    Log.LogMessage("Added '{0}' to archive", file.ItemSpec);
+                    var entry = zip.CreateEntryFromFile(file.ItemSpec, entryName);
+#if NET46
+#elif NETCOREAPP2_0
+                    if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
```

Comment:
I ran into this while trying to build and test this repo on WSL and macOS. `./build.sh` fails without this because files will be extracted as `0000` instead of `0644`.